LaTeX/LaTeX.sublime-syntax

```diff
@@ -702,75 +702,382 @@ contexts:
+      scope: support.function.textbf.latex
+      captures:
+        1: punctuation.definition.backslash.latex
+      push:
+        - textbf-meta
+        - text-bold-body
+        - expect-text-markup-open-brace
+    - match: (\\)textmd{{endcs}}
+      scope: support.function.textmd.latex
+      captures:
+        1: punctuation.definition.backslash.latex
+      push:
+        - textmd-meta
+        - text-normal-body
+        - expect-text-markup-open-brace
+    - match: (\\)textlf{{endcs}}
+      scope: support.function.textlf.latex
+      captures:
+        1: punctuation.definition.backslash.latex
+      push:
+        - textlf-meta
+        - text-normal-body
+        - expect-text-markup-open-brace
+
+  # utility that waits for the brace, and pops the three contexts
+  # we push for each \text... command otherwise
+  expect-text-markup-open-brace:
+    - match: \{
+      scope: punctuation.definition.group.brace.begin.tex
+      pop: true
+    - match: (?=\S)
+      pop: 3
+
+  textnormal-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.function.textnormal.latex
+    - include: immediately-pop
+
+  textbf-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.function.textbf.latex
+    - include: immediately-pop
+
+  textmd-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.function.textmd.latex
+    - include: immediately-pop
+
+  textlf-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.function.textlf.latex
+    - include: immediately-pop
+
+  textsf-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.function.textsf.latex
+    - include: immediately-pop
+
+  textup-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.function.textup.latex
+    - include: immediately-pop
+
+  textsc-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.function.textsc.latex
+    - include: immediately-pop
+
+  textrm-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.function.textrm.latex
+    - include: immediately-pop
+
+  textit-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.function.textit.latex
+    - include: immediately-pop
+
+  texttt-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.function.texttt.latex
+    - include: immediately-pop
+
+  textsl-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.function.textsl.latex
+    - include: immediately-pop
+
+  emph-meta:
+    - meta_include_prototype: false
+    - meta_scope: meta.function.emph.latex
+    - include: immediately-pop
+
+  text-normal-body:
+    - include: brace-group-end
+    - include: main
+
+  text-bold-body:
+    - include: brace-group-end
+    - include: text-bold-select-font-family
+    - include: text-bold-select-font-shape
+    - match: \{
+      scope: punctuation.definition.group.brace.begin.tex
+      push: text-bold-brace-group-body
+    - include: main
+    - match: .
```

Comment:
This pattern causes ST to create a token for each single character.

See how each char of "bold" and "still bold" is highlighted differently.

![grafik](https://user-images.githubusercontent.com/16542113/213877196-6cdd50a8-17be-4953-a3a3-b7084e2c8e1e.png)

A possible solution/improvement is to use a pattern which matches all chars, which the other patterns are not expected to start with.

The following pattern:

```suggestion
    - match: '[^\\{}]+'
```

produces more sane results:

![grafik](https://user-images.githubusercontent.com/16542113/213877340-fb1da26b-a248-4eb5-943a-b0c175c7714f.png)

_Note: Effects the other two occurances in `italic` and `bold-italic` as well._

Reply:
its a bit more complicated than that, e.g. you suggestion would break `\textbf{bold $normal math$}` (which is currently missing from the tests)

Reply:
I'd vote to add tests for some of the more exotic cases as well in general. As those are often root cause of breaking highlighting _(lessons learned)_.

I already was concerned about such cases.

If in doubt we can go with your solution - it's not wrong - just not optimal from performance/memory concumption point of view.

Reply:
`\w+|.`?

Reply:
If all special chars were known those could be added to the char class `[^\\{}%$]+`. Otherwise we could refine your suggestion to something like `.[\s\w]*` maybe.

Reply:
If it was acceptable to also scope keywords `markup.bold`, ... a `meta_content_scope` may be used as well to scope everything within braces accordingly.

Reply:
I think I prefer the more refined match over making everything `markup.bold`. Explicitly opting out only for a few characters might work, because
a) even if they have special meaning in latex, it really hurts only for those for which we do specific processing
b) any sort of fancy latex code (directly) inside text markup is very unlikely. E.g., the `#`, `_`, `^`, `&` characters would be problematic, but I think they aren't actually possible inside well-formed latex, so even if we do get these wrong, nothing too bad would be happening.